### PR TITLE
Stats for each minute

### PR DESCRIPTION
**Pull request by mr-nmh**
Result after run
https://gist.github.com/devnmhoang/8d4a522d365ae7f95d29ef239b490c0e

### CODE REVIEW

File: tx_metric/tx_metric.go [hunk 1]
```diff
@@ -61,12 +74,20 @@ func (tm *TxMetric) Start() error {
...
 	fmt.Println("Total Blocks:", totalBlock)
 	fmt.Println("Total Blocks have 0 Tx:", numberOfBlockHasNoTx)
 	fmt.Println("=> AVG Txs/Block:", float64(totalTx)/float64(totalBlock))
 	fmt.Println("=> TPS:", float64(totalTx)/tm.Duration.Seconds())
+
+	// Calculate stats for each minute
+	if tm.Duration.Minutes() > 1 {
+		fmt.Println("-----------Detail stats for each minute----------------")
+		for index, txs := range minuteStats {
+			fmt.Println("Txs at minute", index+1, ":", txs)
```
Comment (ducquangkstn) [inline tx_metric/tx_metric.go]: `Printf` is recommended

File: tx_metric/tx_metric.go [hunk 2]
```diff
@@ -37,10 +38,17 @@ func (tm *TxMetric) Start() error {
...
-		numberOfBlockHasNoTx = 0
+		totalTx                 = 0
+		totalBlock              int64
+		numberOfBlockHasNoTx    = 0
+		calculateEachMinuteFlag bool
+		minuteStats             = make([]int, int(math.Floor(float64(tm.StopTime-tm.StartTime)/60.0))+1)
 	)
+
+	if tm.Duration.Minutes() > 1 {
+		calculateEachMinuteFlag = true
```
Comment (ducquangkstn) [inline tx_metric/tx_metric.go]:
```suggestion
calculateEachMinuteFlag := tm.Duration.Minutes() > 1
```

File: tx_metric/tx_metric.go [hunk 3]
```diff
@@ -51,7 +59,12 @@ func (tm *TxMetric) Start() error {
 
 		if bl.Time() <= tm.StopTime {
```
Comment (ducquangkstn) [inline tx_metric/tx_metric.go:55]: it is not at the PR, but we can add an early break with condition bl.Time() > tm.StopTime

File: tx_metric/tx_metric.go [hunk 4]
```diff
@@ -51,7 +59,12 @@ func (tm *TxMetric) Start() error {
...
 			fmt.Printf("Found blocknumber %d | Txs: %d\n", i, bl.Transactions().Len())
-			totalTx += bl.Transactions().Len()
+			numberOfTx := bl.Transactions().Len()
+			if calculateEachMinuteFlag {
+				minuteStats[int(math.Floor(float64(bl.Time()-tm.StartTime)/60.0))] += numberOfTx
```
Comment (ducquangkstn) [inline tx_metric/tx_metric.go]: `int(bl.Time()-tm.StartTime)/60` is ok, not need math.Floor

**Review by ducquangkstn — APPROVED** [review]
LGTM. However pls refactor a bit

File: tx_metric/tx_metric.go (hunk 3 above)
Comment (mr-nmh) [reply tx_metric/tx_metric.go:55]: We had 
`} else {
			break
		}`
so I think adding an early break is not necessary.

File: tx_metric/tx_metric.go (hunk 4 above)
Comment (mr-nmh) [reply tx_metric/tx_metric.go]: Updated

File: tx_metric/tx_metric.go (hunk 2 above)
Comment (mr-nmh) [reply tx_metric/tx_metric.go]: Updated

File: tx_metric/tx_metric.go (hunk 1 above)
Comment (mr-nmh) [reply tx_metric/tx_metric.go]: updated

**Pull request merged**